src/main/scala/wandou/avpath/Evaluator.scala

```diff
@@ -16,70 +16,84 @@ import wandou.avpath.Parser.Syntax
 import wandou.avpath.Parser.UnaryExprSyntax
 import wandou.avpath.Parser.PosSyntax
 import java.nio.ByteBuffer
+
 import org.apache.avro.Schema
 import org.apache.avro.Schema.Type
 import org.apache.avro.generic.GenericEnumSymbol
 import org.apache.avro.generic.GenericFixed
 import org.apache.avro.generic.IndexedRecord
+
 import scala.collection.mutable
 
 object Evaluator {
+
   sealed trait Target
+
   final case class TargetRecord(record: IndexedRecord, field: Schema.Field) extends Target
+
   final case class TargetArray(array: java.util.Collection[_], idx: Int, arraySchema: Schema) extends Target
+
   final case class TargetMap(map: java.util.Map[String, _], key: String, mapSchema: Schema) extends Target
 
   private object Op {
+
     case object Select extends Op
+
     case object Update extends Op
+
     case object Delete extends Op
+
     case object Clear extends Op
+
     case object Insert extends Op
+
     case object InsertAll extends Op
+
   }
+
   private sealed trait Op
 
-  final case class Ctx(value: Any, schema: Schema, topLevelField: Schema.Field, target: Option[Target] = None)
+  final case class Ctx(value: Any, name: String, schema: Schema, topLevelField: Schema.Field, path: String, target: Option[Target] = None)
```

Comment:
:/  Can you  comment (or even :gasp: javadoc) what name and path are?

I know: you'd be the first on in avpath to do it!

`path` is the original query, right?  and `name` is the *current* field name (if any) being processed?